packages/aws-cdk-lib/aws-ecr/lib/repository.ts

```diff
@@ -706,6 +714,7 @@ export class Repository extends RepositoryBase {
       imageScanningConfiguration: props.imageScanOnPush !== undefined ? { scanOnPush: props.imageScanOnPush } : undefined,
       imageTagMutability: props.imageTagMutability || undefined,
       encryptionConfiguration: this.parseEncryption(props),
+      emptyOnDelete: props.emptyOnDelete,
```

Comment:
We should ignore `autoDeleteImages` if `emptyOnDelete` is set [here](https://github.com/aws/aws-cdk/blob/03a5ecdfce0a17dd15097806571db237aaa667c0/packages/aws-cdk-lib/aws-ecr/lib/repository.ts#L726):
```
if (props.emptyOnDelete === undefined && props.autoDeleteImages) {
```
(I think that setting `emptyOnDelete = false` should override `autoDeleteImages = true`, but feel free to comment on this).

Also:
* This will need some extra unit tests
* Does `emptyOnDelete = true` requires `removalPolicy === RemovalPolicy.DESTROY`? I wasn't able to find documentation on this

Reply:
- I'm fine with only using `autoDeleteImages` if `emptyOnDelete` is `undefined`.
- `emptyOnDelete = true` does not require `removalPolicy === RemovalPolicy.DESTROY`

Made the change and added tests, please review 🙂